### ISSUE TITLE
[travis] Travis is not posting build statuses to PRs

### ISSUE THREAD

**Issue by TurboTurtle** [sosreport/sos#2246]
For the last few weeks, Travis has not been posting build states to newly opened or updated PRs, though it does appear the builds are actually running.

The most recent PR push has led to a new `flake8` failure getting into master because of this, because I did not notice the lack of making a regex string with escape sequences a raw string and the nromal catch from Travis was not there.

That is easy enough to fix, but I'd rather not be chasing these via build failure notifications after pushes to master.

@BryanQuigley can you look into the Travis integration? I'm not privy to how that is setup, but I don't see anything within the sosreport org or the sos repo specifically for it (and at the same time I don't see anything that would outright block it from posting).

More generally - should we start looking at alternatives to Travis for our CI testing? Jenkins? CircleCI?

**Comment by BryanQuigley** [comment]
Github Actions seem like they can do a lot these days

I am surprised that as you said the builds are running - https://travis-ci.org/github/sosreport/sos/pull_requests.  

I don't understand why it worked for me in my test PR.

A couple of posts indicate that deauthorizing and then reauthorizing Travis might help: https://travis-ci.community/t/github-pr-is-being-built-but-result-is-not-shown/7025.

**Comment by TurboTurtle** [comment]
Well, since your test PR it seems to be back and functioning. I didn't do a deauthorization, so I'm not sure what exactly changed.

I looked into GH Actions, and it looks like it could be a good alternative (and likely won't have these syncing issues). I only stopped researching it since, if we make the plunge to change CI platforms, I'd like to be able to include Fedora/RHEL testing as well. With self-hosted runners this is possible, however any fork of this repo has access to those runners as well (which makes absolutely no sense to me).

Though, one thing we could do is transition to GH Actions for the Ubuntu based tests, and simply have a secondary integration (CircleCI looks promising) for the Fedora stuff.

TL;DR - if you like GH Actions over Travis, I have no objections.

**Comment by BryanQuigley** [comment]
I don't see a clear path moving to anything else to get Fedora/RHEL testing online (aside from using runners as you mentioned).  My impression of most of them are that it ends up running in a Docker container on Ubuntu which is likely useful for most other applications.

It would be some work to move to GH actions, but I don't see enough benefit it it's working again.

**Comment by TurboTurtle** [comment]
I've spoken with Circle and for a couple reasons its a non-starter for sos.

That being said, I've also been in contact with the podman maintainers, who use Cirrus quiet extensively and it looks promising for us here. I need to square away the billing for spinning up on GCP, but I should be able to start testing a CI setup with Cirrus shortly.
